aten/src/ATen/core/ATenDispatch.h

```diff
@@ -15,6 +16,26 @@
 
 namespace at {
 
+namespace impl {
+
+// Take a TensorTypeSet for a Tensor, and combine it with the current thread
+// local set of inclusions and exclusions (not yet implemented, coming soon!)
+// to determine what the actual dispatch TensorTypeId should be.  Unlike
+// Tensor::type_set(), the value of this on a tensor can change depending
+// on TLS.
+//
+// NB: I didn't make this take a Tensor to avoid header include shenanigans.
+//
+// TODO: I'm not sure if this should live in this header or not; the operant
+// question is whether or not we have access to all the relevant TLS at this
+// point.
+static inline TensorTypeId dispatchTypeId(TensorTypeSet ts) {
```

Comment:
why is this a separate function hidden away here? wouldn't it either be a method in Tensor or just in ts.highestPriorityTypeId?

Reply:
Separate header probably makes more sense because you can include it from globalATenDispatch **and** the c10 dispatcher.

Reply:
In the later PRs, I determined that yes, I do in fact have access to all the TLS :)

Otherwise, there's no particular reason I put it here, except that this was where I needed the first site. The function *should* be hidden away, since only the dispatchers really have any legitimate use for it.

Reply:
If it turns out I need to move it for c10 dispatcher, I'll move it, but for now I intend to park it here.